runner/run.go

```diff
@@ -55,6 +59,14 @@ func Run(ctx context.Context, wg *sync.WaitGroup, globalCollectionOpts state.Col
 				logger.PrintError("Failed to initialize OpenTelemetry tracing provider, disabling exports: %s", err)
 			}
 		}
+
+		if cfg.DbUseIamAuth && cfg.SystemType == "google_cloudsql" && driverCleanup == nil {
+			driverCleanup, err = pgxv5.RegisterDriver("cloudsql-postgres", cloudsqlconn.WithIAMAuthN())
```

Comment:
I can't really do `defer driverCleanup()` here as the cleanup needs to happen at the caller, main.go.
Since there was already a nice `shutdown` func, I piggybacked it and called `driverCleanup()` there.
Registering driver needs to happen only one time per the process, therefore it lives here and doing the nil check of driverCleanup.

https://github.com/GoogleCloudPlatform/cloud-sql-go-connector?tab=readme-ov-file#using-the-dialer-with-databasesql